### PR TITLE
Postgres Writer #3 : Move transaction and batching

### LINKED ISSUE #566: [parser refactor] Split PostgresWritingRecordParsedItemHandler out of RecordFileLogger
Move all connection and PreparedStatement logic out of RecordFileLogger
- `PostgresWritingRecordParsedItemHandler` will
  - implement `RecordParsedItemHandler`
  - own connection, setup PreparedStatements.
  - handle batching logic
  - Do `f_file_create` (in `initFile`) and `f_file_complete` (in `completeFile`)
- Keep all Transaction/TransactionRecord parsing logic in `RecordFileLogger` (to be renamed separately to keep PRs small)

### PR DESCRIPTION
**Pull request by apeksharma**
**Detailed description**:
Move transaction and batching to `PostgresWritingRecordParserItemHandlerTest`.

Followups from previous reviews:
- Add NotImplementedException to methods missing logic
- share code between two insertFile*() functions

Signed-off-by: Apekshit Sharma <apekshit.sharma@hedera.com>

**Which issue(s) this PR fixes**:
Partially fixes #566 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/RecordFileLoggerCryptoTest.java [hunk 1]
```diff
@@ -137,29 +137,6 @@ void cryptoCreate() throws Exception {
         verifyRepoCryptoTransferList(record);
     }
 
-    @Test
```
Comment (apeksharma) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/RecordFileLoggerCryptoTest.java]: not needed now that there's test for batching in `PostgresWritingRecordParserItemHandlerTest`.
Even before, it wasn't testing batching correctly.
Since the test always commits explicitly by calling `completeFile` and then only checks for number of rows in the table, it would pass even if batching was not working correctly (verified it locally on master branch).

**Comment by apeksharma** [comment]
ready for review.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWritingRecordParsedItemHandler.java [hunk 2]
```diff
@@ -36,18 +37,31 @@
...
     private PreparedStatement sqlInsertNonFeeTransfers;
     private PreparedStatement sqlInsertFileData;
     private PreparedStatement sqlInsertContractResult;
     private PreparedStatement sqlInsertLiveHashes;
     private PreparedStatement sqlInsertTopicMessage;
+    private final PostgresWriterProperties properties;
 
     void initSqlStatements(Connection connection) throws ParserSQLException {
         try {
+            sqlInsertTransaction = connection.prepareStatement("INSERT INTO t_transactions"
```
Comment (Nana-EC) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWritingRecordParsedItemHandler.java]: So my thought here was that if the PreparedStatement could be retrieved by test code then we could eventually down ingestion only tests for performance, and we'd avoid test code and product code getting out of sync.

Comment (apeksharma) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWritingRecordParsedItemHandler.java]: Discussed with nana a bit, posting summary.
The goal here is, if these statements are repeating in our codebase in multiple places, then we want to modularize these. We'll discuss it again tomorrow, and can be done independently.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordParserProperties.java [hunk 3]
```diff
@@ -60,6 +60,8 @@
     // bytes on the t_transaction table
     private boolean persistTransactionBytes = false;
 
+    private final PostgresWriterProperties postgresWriter;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordParserProperties.java]: Why is this added here if it is a standalone bean used in a standalone fashion in the handler?

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Data;
+import javax.inject.Named;
+
+@Data
+@Named
+public class PostgresWriterProperties {
+    /**
+     * PreparedStatement.executeBatch() is called after every batchSize number of transactions from record stream file.
+     */
+    private int batchSize = 100;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java]: Needs `@Min(1)`

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.hedera.mirror.importer.parser.record;
+
+import lombok.Data;
+import javax.inject.Named;
+
+@Data
+@Named
+public class PostgresWriterProperties {
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java]: Missing `@ConfigurationProperties("hedera.mirror.parser.record.postgresql")` and `@Validated`. `@Named` can be removed as it's auto-scanned.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileLogger.java [hunk 6]
```diff
@@ -740,28 +693,21 @@ private static void insertConsensusTopicMessage(ConsensusSubmitMessageTransactio
...
-                        .getFileNum() < 1000)) {
-            byte[] contents = transactionBody.getContents().toByteArray();
+                (parserProperties.isPersistSystemFiles() && fileID.getFileNum() < 1000)) {
             postgresWriter.onFileData(new FileData(consensusTimestamp, contents));
         }
     }
 
     private static void insertFileAppend(long consensusTimestamp, FileAppendTransactionBody transactionBody)
             throws IOException {
-        if (parserProperties.isPersistFiles() ||
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileLogger.java]: Why was this check removed?

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java (hunk 4 above)
Comment (apeksharma) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java]: done on followup.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java (hunk 5 above)
Comment (apeksharma) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/PostgresWriterProperties.java]: done on followup.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileLogger.java (hunk 6 above)
Comment (apeksharma) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileLogger.java]: moved to `insertFileData` fn.
networkAddressBook update happens out of the check now
